reframe/frontend/cli.py

```diff
@@ -410,15 +411,27 @@ def main():
 
     # Setup the check loader
     if options.checkpath:
-        load_path = []
+        load_path = OrderedSet()
         for d in options.checkpath.split(':'):
             d = os_ext.expandvars(d)
             if not os.path.exists(d):
                 printer.warning("%s: path `%s' does not exist. Skipping..." %
                                 (argparser.prog, d))
                 continue
 
-            load_path.append(d)
+            d = os.path.abspath(os.path.realpath(d))
+            if d not in load_path:
+                load_path.add(d)
+
+        if options.recursive:
+            for d in list(load_path):
+                d_parent = os.path.dirname(d)
+                while d_parent not in {'', '/'}:
```

Comment:
Why do you check for the empty string? You have converted all paths to absolute.

Reply:
I guess you have  put it for paths ending in `/`. You should normalize them before processing them.